low-level/sensors/impl/centos_7/disk_monitor.py

```diff
@@ -1040,14 +1056,20 @@ def _update_drive_faults(self):
 
         for object_path in self._drives.keys():
 
-            # To handle case when drvie is removed, but interface_removed function is not yet
+            # To handle case when drive is removed, but interface_removed function is not yet
             # called, so drive will be still in self._drives, but smartctl command will fail as
             # device is removed.
             if not self._drives[object_path]["node_disk"]:
                 cmd = f"sudo smartctl -d scsi -H {self._drive_by_device_name[object_path]} --json"
             else:
                 cmd = f"sudo smartctl -H {self._drive_by_device_name[object_path]} --json"
-            response, _, _ = self._run_command(cmd)
+            response, err, retcode = self._run_command(cmd)
+            if retcode != 0 and err:
+                self._iem.iem_fault("SMARTMONTOOL_ERROR")
+                return

```

Comment:
yes, return is required because in case of smartctl cmd failure, 'response' will be none, and  if we proceed further we may get errors and sensor thread will be shutdown.